### ISSUE TITLE
First impression: No obvious feedback when opening trace

### ISSUE THREAD

**Issue by tahini** [lttng/lttng-scope#74]
I just tried scope. File -> Open... yeah! I chose a kernel dir and clicked Open, because I thought that was right (I didn't read the doc first, and I _was_ right, so it's intuitive, good!).

Of course, I chose one of the biggest trace I have (I want to _test_ scope :p). And I thought I had it wrong, nothing happened, until I noticed the little progress bar at the bottom right. Scope decided not to show anything before the end of the analysis, that's fine, but for big traces, there should be something to make it more obvious, maybe if all views had a visual indication that it's waiting for data, or at least have the project appear on the left project explorer.

**Comment by tahini** [comment]
As a follow-up of this issue, say I have a trace opened and want to open another [large] one, the first one stays opened until the data for the second is available. I think if there's only one trace opened at a time, it should close the first one before opening the next. The user will not think she is seeing the new trace, while it is still the previous one.

**Comment by alexmonthy** [comment]
> File -> Open... yeah!

That's going to change pretty soon, unfortunately! See https://github.com/lttng/lttng-scope/issues/77#issuecomment-379037767.

> because I thought that was right (I didn't read the doc first, and I was right, so it's intuitive, good!).

*Exactly*, users shouldn't have to read documentation at all, that's one of the core design goals of this project! Just like iPads don't even have a manual in the box (so I've heard, never owned an Apple device, hah).

> until I noticed the little progress bar at the bottom right

Yeah that's how you're supposed to know... When I open a large document in Word/Libreoffice this is what it looks like too.

Perhaps right now it is a bit hard to see because of all the boxes and panes in the empty UI. Eventually we won't show anything in the central view at first, the widgets placed there will be determined by what is found in the traces, or added/removed manually by the user. Maybe once it looks like that it will be easier to notice that the application is currently working?

> maybe if all views had a visual indication that it's waiting for data

Technically the view is not waiting for data, it's the application that is currently doing a project-switching operation. At some point we will probably have a way to run analyses "lazily", for example only once the user opens a given widget. At that point then yes, the widget will be in a "waiting for data" state. But when it's the whole application doing a project-switching (which includes what we could call "automatic analyses") then it should be shown differently. The widgets shouldn't even exist at that point, like I mentioned earlier.

I thought of a full-window overlay showing that the application is working, but most window manager nowadays will darken an application window when it thinks it's hung, so said overlay could be mistaken for that.

> As a follow-up of this issue, say I have a trace opened and want to open another [large] one, the first one stays opened until the data for the second is available.

That's a good point, in fact what happens underneath is that we close the first project and then open the new one. These actions should be reflected in the UI, so we should remove all the data from the previous project first, and then switch to the new one.

**Comment by alexmonthy** [comment]
Another point we discussed recently:

While a project-switching operation is ongoing and the user cannot interact with anything, we open a  dialog window similar to the current one showing the progress of the ongoing tasks, except that one would be *modal*. Once the project is ready, this window closes and the user can now interact with the UI.

As for eventual "on demand" tasks, we could show the progress of the required task(s) directly into the widgets being loaded. Maybe we won't even need another separate task progress window at that point.

**Comment by alexmonthy** [comment]
>> As a follow-up of this issue, say I have a trace opened and want to open another [large] one, the first one stays opened until the data for the second is available.

> That's a good point, in fact what happens underneath is that we close the first project and then open the new one. These actions should be reflected in the UI, so we should remove all the data from the previous project first, and then switch to the new one.

FYI, this should be fixed now with https://github.com/efficios/jabberwocky/commit/38a487caef6a28d1655f248a7872bf1ecb27ed74.
